src/models/projectConfiguration.ts

```diff
@@ -19,177 +20,237 @@ export enum Language {
+  region: string;
+  sdk: SdkConfiguration;
+  classes: ClassConfiguration[];
+
+  constructor(
+    name: string,
+    region: string,
+    sdk: SdkConfiguration,
+    classes: ClassConfiguration[]
+  ) {
+    this.name = name;
+    this.region = region;
+    this.sdk = sdk;
+    this.classes = classes;
   }
 
-  static async create(configurationFileContent: any): Promise<ProjectConfiguration> {
+  static async create(
+    configurationFileContent: any
+  ): Promise<ProjectConfiguration> {
     if (!configurationFileContent.name) {
-      throw new Error("The name property is missing from the configuration file.")
+      throw new Error(
+        "The name property is missing from the configuration file."
+      );
     }
 
     if (!configurationFileContent.sdk.path) {
-      throw new Error("The sdk.path property is missing from the configuration file.")
+      throw new Error(
+        "The sdk.path property is missing from the configuration file."
+      );
     }
 
-    const language: string = configurationFileContent.sdk.language
+    const language: string = configurationFileContent.sdk.language;
 
     if (!language || !Language[language as keyof typeof Language]) {
-      throw new Error("The sdk.language property is invalid.")
+      throw new Error("The sdk.language property is invalid.");
+    }
+
+    if (
+      (Language[
+        configurationFileContent.sdk.language as keyof typeof Language
+      ] == Language.js ||
+        Language[
+          configurationFileContent.sdk.language as keyof typeof Language
+        ] == Language.ts) &&
+      !JsRuntime[configurationFileContent.sdk.runtime as keyof typeof JsRuntime]
+    ) {
+      throw new Error("The sdk.runtime property is invalid.");
     }
 
-    if ((Language[configurationFileContent.sdk.language as keyof typeof Language] == Language.js ||
-      Language[configurationFileContent.sdk.language as keyof typeof Language] == Language.ts) &&
-      !JsRuntime[configurationFileContent.sdk.runtime as keyof typeof JsRuntime]) {
-        throw new Error("The sdk.runtime property is invalid.")
-      }
-    
     const sdk = new SdkConfiguration(
       Language[configurationFileContent.sdk.language as keyof typeof Language],
       JsRuntime[configurationFileContent.sdk.runtime as keyof typeof JsRuntime],
       configurationFileContent.sdk.path
-    )
+    );
 
-    const unparsedClasses: any[] = configurationFileContent.classes
+    const unparsedClasses: any[] = configurationFileContent.classes;
 
     if (!unparsedClasses) {
-      throw new Error("The configuration file should contain at least one class.")
+      throw new Error(
+        "The configuration file should contain at least one class."
+      );
     }
 
-    const classes = await Promise.all(unparsedClasses.map((c) => ClassConfiguration.create(c)))
+    const classes = await Promise.all(
+      unparsedClasses.map((c) => ClassConfiguration.create(c))
+    );
+
+    if (configurationFileContent.region) {
```

Comment:
I think I would put this check inside in the `ProjectConfiguration` because we have there all the verifications.